packages/runtime/lib/start.js

```diff
@@ -70,8 +70,51 @@ async function startWithConfig (configManager) {
     async restart () {
       worker.postMessage({ msg: 'plt:restart' })
       await once(worker, 'message') // plt:restarted
+    },
+    async getServicesTopology () {
+      worker.postMessage({ msg: 'plt:get-topology' })
+      const [msg] = await once(worker, 'message')
+
+      const res = JSON.parse(msg.res)
+      if (res.error) {
+        throw new Error(res.error)
+      }
+
+      return res
+    },
+    async getServiceState (id) {
+      worker.postMessage({ msg: 'plt:get-state', params: { id } })
+      const [msg] = await once(worker, 'message')
+
+      const res = JSON.parse(msg.res)
+      if (res.error) {
+        throw new Error(res.error)
+      }
+
+      return res
+    },
+    async getServiceConfig (id) {
+      worker.postMessage({ msg: 'plt:get-config', params: { id } })
+      const [msg] = await once(worker, 'message')
+
+      const res = JSON.parse(msg.res)
+      if (res.error) {
+        throw new Error(res.error)
+      }
+
+      return res
+    },
+    async startService (id) {
+      worker.postMessage({ msg: 'plt:start-service', params: { id } })
+      await once(worker, 'message')
+    },
+    async stopService (id) {
+      worker.postMessage({ msg: 'plt:stop-service', params: { id } })
+      await once(worker, 'message')
     }
```

Comment:
Can you add an API to ask for the OpenAPI and GraphQL schemas for a given service?

Reply:
IMHO we should think about adding a reverse proxy directly inside the runtime (or as a hidden runtime nanoservice) and expose it in a separate port. WDYT?

Reply:
Take a look at https://github.com/platformatic/platformatic/pull/1096/files#r1231206659, it might be preferable to support inject() across the worker boundary.